tests/src/test/java/alluxio/master/journal/ufs/UfsJournalLogWriterTest.java

```diff
@@ -178,6 +197,229 @@ public void writeJournalEntryRotate() throws Exception {
     }
   }
 
+  /**
+   * Tests that (1) {@link UfsJournalLogWriter#mJournalOutputStream} is reset when an exception
+   * is thrown, and (2) the {@link UfsJournalLogWriter} recovers during the next write. It should
+   * write out all entries, including the one that initially failed.
+   */
+  @Test
+  public void recoverFromUfsFailure() throws Exception {
+    long startSN = 0x10;
+    UfsJournalLogWriter writer = new UfsJournalLogWriter(mJournal, startSN);
+    final int numberOfEntriesWrittenBeforeFailure = 10;
+    long nextSN = writeJournalEntries(writer, startSN, numberOfEntriesWrittenBeforeFailure);
+    // Create a mock DataOutputStream.
```

Comment:
Avoid comments that just repeat what the code says.

Reply:
Done.